### PR TITLE
[feat]: [CDS-87897] : new schema v1 for service entity crud

### PR DESCRIPTION
**Pull request by tchaurasiya**
Derived from
https://harness.atlassian.net/wiki/spaces/~5ce42d90c9167e0d6ea29342/pages/21583530813/Service+Environment+Yaml+Simplification#Final-Service-Yaml

service type supported: kubernetes

manifest supported: k8s, values
manifest store supported : github

artifact supported: docker

config-file store - harness

### CODE REVIEW

**Review by brijeshdhakar — DISMISSED** [review]

File: v1/serviceEntity/k8s-manifest.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      - github
+    values:
+      oneOf:
+      - type: array
+        items:
+          type: string
+      - type: string
+        pattern: "(<\\+.+>.*)"
+        minLength: 1
+    skip_resource_versioning:
```
Comment (shalini-agr) [inline v1/serviceEntity/k8s-manifest.yaml:21]: can we rename this to `skip_versioning`?

File: v1/serviceEntity/stores/docker-hub-artifact-config.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+allOf:
+- $ref: basic-artifact-store-config.yaml
+- type: object
+  required:
+  - connector
+  - location
+  properties:
+    connector:
+      type: string
+    location:
```
Comment (shalini-agr) [inline v1/serviceEntity/stores/docker-hub-artifact-config.yaml:11]: please add description of location field

File: v1/serviceEntity/stores/github-store-config.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+title: GithubStoreConfig
+allOf:
+- $ref: basic-manifest-store-config.yaml
+- type: object
+  required:
+  - connector
+  - location
+  properties:
+    location:
```
Comment (shalini-agr) [inline v1/serviceEntity/stores/github-store-config.yaml:9]: please add description for location field

File: v1/serviceEntity/stores/harness-store.yaml [hunk 4]
```diff
@@ -0,0 +1,17 @@
+title: HarnessStore
+allOf:
+- $ref: basic-config-file-store-config.yaml
+- type: object
+  properties:
+    file:
```
Comment (shalini-agr) [inline v1/serviceEntity/stores/harness-store.yaml:6]: please add description for file field

File: v1/serviceEntity/config-file.yaml [hunk 5]
```diff
@@ -0,0 +1,21 @@
+title: ConfigFile
+type: object
+required:
+- id
```
Comment (brijeshdhakar) [inline v1/serviceEntity/config-file.yaml:4]: Do we need to make the id required? If its not needed then lets make it non required.

File: v1/serviceEntity/manifest-wrapper.yaml [hunk 6]
```diff
@@ -0,0 +1,12 @@
+title: ManifestsWrapper
+type: object
+required:
+- sources
+properties:
+  primary:
```
Comment (brijeshdhakar) [inline v1/serviceEntity/manifest-wrapper.yaml:6]: What does primary mean in manifests?

File: v1/serviceEntity/config-file.yaml (hunk 5 above)
Comment (tchaurasiya) [reply v1/serviceEntity/config-file.yaml:4]: Yes @brijeshdhakar in override we override config file based on id.

File: v1/serviceEntity/manifest-wrapper.yaml (hunk 6 above)
Comment (tchaurasiya) [reply v1/serviceEntity/manifest-wrapper.yaml:6]: in helm deployment we have primary manifest concept.
There user is allowed to give multiple manifest in helm deployment.

File: v1/serviceEntity/stores/github-store-config.yaml (hunk 3 above)
Comment (tchaurasiya) [reply v1/serviceEntity/stores/github-store-config.yaml:9]: done.

**Review by shalini-agr — APPROVED** [review]

**Review by archit-harness — APPROVED** [review]

**Pull request merged**